src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -6862,18 +6876,60 @@ void ExportMusicXml::keysigTimesig(const Measure* m, const Part* p)
         }
     }
 
-    TimeSig* tsig = 0;
+    // search all staves for non-generated time signatures
+    std::map<staff_idx_t, TimeSig*> timesigs;   // map staff to time signature
     for (Segment* seg = m->first(); seg; seg = seg->next()) {
```

Comment:
A small change, this could be 
```suggestion
    for (Segment* seg = m->first(); seg; seg = seg->next(SegmentType::TimeSig)) {
```